libs/designer/src/lib/core/queries/connector.ts

```diff
@@ -181,6 +181,45 @@ export const getLegacyDynamicTreeItems = async (
   return response;
 };
 
+export const getDynamicTreeItems = async (
+  connectionId: string,
+  connectorId: string,
+  operationId: string,
+  parameterAlias: string | undefined,
+  parameters: Record<string, any>,
+  dynamicState: any
+): Promise<TreeDynamicValue[]> => {
+  const queryClient = getReactQueryClient();
+  const service = ConnectorService();
+
+  const values = await queryClient.fetchQuery(
+    [
+      'dynamictreeitems',
+      connectionId.toLowerCase(),
+      connectorId.toLowerCase(),
+      operationId?.toLowerCase(),
+      getParametersKey(parameters).toLowerCase(),
+    ],
+    () => service.getTreeDynamicValues(connectionId, connectorId, operationId, parameterAlias, parameters, dynamicState)
+  );
+
+  if (values && values.length) {
+    return values.map((treeDynamicValue: any) => {
+      return {
+        displayName: treeDynamicValue.displayName,
+        isParent: treeDynamicValue.isParent,
+        value: {
+          id: treeDynamicValue.dynamicState?.selectionState?.id,
+          [dynamicState.dynamicExtension.browse.itemValuePath]: treeDynamicValue.value,
+          [dynamicState.dynamicExtension.browse.itemFullTitlePath]: treeDynamicValue.fullyQualifiedDisplayName,
+        },
+      };
+    });
+  }
+
+  return values;
+};
```

Comment:
the connectors [x-ms-dynamic-tree].browse.itemFullTitlepath value should be equal to the dynamicitem returned from the call[itemFullTitlePath] if that is not the case then itemFullTitlePath should never be set in the connector because noOne would be using it in the client.

Something is redundant here based on the metadata.
And please do all value modifications in your srvice it should not be done here, this is comon code and LA might not send results like this.

Reply:
Yes, I'm trying to move the value modifications to our service but am unable to get things working end to end without a change to getPickerCallbacks function and need your help here.

For more context:

During initialization of action, we don't have itemFullTitlepath etc. values in dynamicState.browse. They exist in dynamicState.dynamicExtension.browse instead. This is not a big problem. I can add the values to dynamicState.browse too on our end, so that filePickerInfo.browse has these values:
![image](https://github.com/Azure/LogicAppsUX/assets/8736762/3374e7c4-da44-4b02-876d-ee2b117d31e5)

But the problem is when I'm trying to get the values from tree dynamic call and use those values to set the display in parameter (getPickerCallbacks being used to set the value for display), filePickerInfo values don't match the propertyPath in selectedValue.

For example:

For the call to get Root for folder picker, our tree dynamic call returns these values:
![image](https://github.com/Azure/LogicAppsUX/assets/8736762/8d4ee20a-ea6d-4e2a-829b-ab600c0ffe9b)

We want to use the value and fullyQualifiedDisplayName from here to display the value in parameter.

Result of get dynamic tree call on LAUX side (I can update the dynamic extension browse parameters here in my service to use fullyQualifiedDisplayName instead):
![image](https://github.com/Azure/LogicAppsUX/assets/8736762/9a783867-2f19-453d-bf6e-28d9535944d9)

But this doesn't seem to help with filePickerInfo.browse which is used to display the value in parameter panel.

It might be easier to show and solve if you have 10 mins for a call.

Reply:
Basically, is it possible to update filePickerInfo.browse based on this tree dynamic call or is it okay to use "dependency.definition?.extension?.dynamicState?.dynamicExtension?.browse.itemFullTitlePath" or similar for fallback in the getPickerCallbacks function?

Reply:
If you are already updating the returned item to have 'Path': '/' , now this returned item is the selectedItem in pickerCallbacks
then in the picker callbacks - when the call comes for getdisplayValueFromSelectedItem it will get 
selectedItem['Path'] because we do selectedItem[dependency.filepickerInfo.browse.itemFullPath]

dynamic tree extensions are supposed to contain all metadata inside open/browse.

extension is the x-ms-dynamic-tree which is prsent in parameter schema in manifest, i dont know where the confusion is

Reply:
Actually, I think I found a way to pass it correctly so that filePickerInfo works. I'll try a couple more things first.

Reply:
Fixed this to only return values.